airflow/providers/amazon/aws/operators/redshift_cluster.py

```diff
@@ -458,11 +460,15 @@ def __init__(
         *,
         cluster_identifier: str,
         aws_conn_id: str = "aws_default",
+        deferrable: bool = False,
+        poll_interval: int = 10,
```

Comment:
Just wondering how we would resolve in case if we want use more generic parameters `wait_for_completion` as well as `waiter_delay` and `waiter_max_attempts`.

@ferruzzi @vincbeck Are any plans for migrate other AWS Operators to use waiters?

Reply:
> Are any plans for migrate other AWS Operators to use waiters?

Personally, I think I'd like to see all operators which implement a `wait_for_completion` block to use a waiter (either an official one or a custom one) but it's a matter of time.  I am working on moving the EMR operators over as I get time but it's not a huge priority (yet?) and maybe it's better suited to a community project with a discussion to track the progress like we've done in the past?

Reply:
Any chance that`botocore` will have native asyncio support 🤔 Or it is just some kind of surprise 🤣

Reply:
I can't speak for the botocore team on that, I don't have any contact with that team, so I wouldn't know any better than you on that one.  I'm not aware of anything in that regard but I wouldn't really know.  Syed is working on researching deferrable operators to start making the full suite of AWS operators deferrable though, so out of our team I'd say he's the closest to an expert on aiobotocore and related stuff.

Reply:
It's been requested for a looong time: https://github.com/boto/botocore/issues/458
We reached out to them and they said Botocore will eventually support async by default, but it isn't coming any time soon.

Reply:
Yeah I think we do not have native support soon.
Just hope that point for some inside 🤣